chat-plugins/mafia.js

```diff
@@ -1,919 +1,916 @@
+			if (query[0] === 'roles') {
+				buf += `<h3>Setting the roles</h3>`;
+				buf += `<p>To set the roles, use /mafia setroles [comma seperated list of roles] in ${room.title}.</p>`;
+				buf += `<p>The following key words determine a role's alignment (If none are found, the default alignment is town):</p>`;
+				buf += `<p style="font-weight:bold">${Object.keys(MafiaData.alignments).map(a => { return `<span style="color:${MafiaData.alignments[a].color || '#FFF'}">${MafiaData.alignments[a].name}</span>`; }).join(', ')}</p>`;
 			} else {
-				flavorText += 'You sit down with the other members of the mafia';
-				for (const [i, otherVoter] of otherVoters.entries()) {
-					if (i === otherVoters.length - 1) {
-						flavorText += ' and ';
-					} else {
-						flavorText += ', ';
-					}
-					flavorText += Chat.escapeHTML(otherVoter.name);
-				}
+				buf += `<p><a class="button" href="/view-mafia-${room.id}-roles" target="replace">How to setup roles</a></p>`;
 			}
-
-			voter.voteWindow(voter.class.image, flavorText);
-		}
-	}
-
-	townMeeting() {
-		this.meeting = 'town';
-		this.currentVote = new Map();
-
-		for (let i in this.players) {
-			let player = this.players[i];
-			player.validVotes = this.players;
-
-			player.voteWindow(player.class.image, 'Outraged over the mafia\'s activity in town, the people decide to lynch a person they suspect of being involved with the mafia.');
+			buf += `<p style="font-weight:bold;">Sub List: ${room.game.subs.join(', ')}</p>`;
+			buf += `<p style="font-weight:bold;">Players who are requesting a sub: ${room.game.requestedSub.join(', ')}</p>`;
 		}
-	}
-
-	gameEvent(event, timer) {
-		this.executionOrder = [];
-
-		for (let i in this.players) {
-			let player = this.players[i];
-			if (player.class[event]) {
-				player.event(event);
+		if (!isPlayer && !isHost) {
+			if (!room.game.started) {
+				buf += `<p><button class="button" name="send" value="/mafia join ${room.id}">Join game</button></p>`;
+			} else if (room.game.subs.includes(user.userid)) {
+				buf += `<p><button class="button" name="send" value="/mafia sub ${room.id}, out">Cancel sub request</button></p>`;
+			} else {
+				buf += `<p><button class="button" name="send" value="/mafia sub ${room.id}, in">Join the game as a sub</button></p>`;
 			}
 		}
-
-		this.executionOrder.sort((a, b) =>
-			b.class[event].priority - a.class[event].priority
-		);
-
-		if (this.executionOrder.length || this.currentVote) {
-			this.setTimer(timer);
-		} else {
-			this.setTimer(0.25);
-		}
-	}
-}
+		buf += `</div>`;
+		return buf;
+	},
+};
 
 exports.commands = {
 	mafia: {
-		create: 'new',
-		new: function (target, room, user) {
-			if (!this.can('game', null, room)) return false;
-			if (!room.mafiaEnabled) return this.errorReply("Mafia is disabled for this room.");
+		host: function (target, room, user) {
+			if (!room.mafiaEnabled) return this.errorReply(`Mafia is disabled for this room.`);
```

Comment:
Semi unrelated: You could create a custom room setting specifically for when in Mafia to toggle this with

Reply:
oh thanks for reminding me, I was planning on adding that to room-settings.js